src/Native/LibTorchSharp/THSTorch.cpp

```diff
@@ -323,4 +323,114 @@ double THSSpecial_erf_scalar(const double x)
 double THSSpecial_erfc_scalar(const double x)
 {
     return erfc(x);
-}
\ No newline at end of file
+}
+
+bool THSTorch_is_torch_function_mode_enabled()
+{
+    return at::impl::torch_function_mode_enabled(); //https://github.com/pytorch/pytorch/blob/2c91e13afc6edcfe0a0e6189a88aae4ecbbf3516/torch/csrc/autograd/init.cpp#L911
+}
+
+bool THSTorch_is_autocast_cache_enabled()
+{
+    return at::autocast::is_autocast_cache_enabled();
+}
+
+bool THSTorch_is_autocast_cpu_enabled()
+{
+    return at::autocast::is_cpu_enabled();  //https://github.com/pytorch/pytorch/blob/2c91e13afc6edcfe0a0e6189a88aae4ecbbf3516/torch/csrc/autograd/init.cpp#L523
+}
+
+bool THSTorch_is_autocast_gpu_enabled()
+{
+    return at::autocast::is_enabled(); //https://github.com/pytorch/pytorch/blob/2c91e13afc6edcfe0a0e6189a88aae4ecbbf3516/torch/amp/autocast_mode.py#L363
+}
+bool THSTorch_is_autocast_xpu_enabled()
+{
+    return at::autocast::is_xpu_enabled();
+}
+bool THSTorch_is_autocast_hpu_enabled()
+{
+    return at::autocast::is_hpu_enabled();
+}
+
+#if (TORCH_VERSION_MAJOR ==2 && TORCH_VERSION_MINOR > 0)
+bool THSTorch_is_autocast_ipu_enabled()
+{
+    return at::autocast::is_ipu_enabled();
+}
+
+bool THSTorch_is_autocast_xla_enabled()
+{
+    return at::autocast::is_xla_enabled();
+}
+
+#endif
+
+int8_t THSTorch_get_autocast_cpu_dtype()
+{
+    return (int8_t)at::autocast::get_autocast_cpu_dtype();
+}
+
+int8_t THSTorch_get_autocast_gpu_dtype()
+{
+    //TODO: Implement AUTOCAST AMP AND GRADSCALER
```

Comment:
Is this a work-in-progress PR, or something you're submitting for approval and merging? If the latter, then please create an issue to track "to do" items and add some unit tests.

Reply:
U can merging this if you want, this dont break anything (far as I know). But may useful for someone who want use that autocast function manually. My idea and plan is to make AMP, GradScaler, these modules use the functions I added.
Thank, I will try take into account about issue "to do" and unit tests. Sorry.